harness-requirements.typ

```diff
@@ -1,22 +1,57 @@
 = Test Harness Requirements
-
+//make the formatting of requirements pretty
 In order for the test harness to be effective, it must be able to cover as much code of the balancer as possible. In order to do so, typical unit tests are insufficient to test complex cases like network fragmentation, adding and removing monoliths, etc.
 
 Requirement priority is in ascending order, with P1 being the highest priority and P5 being the lowest.
 
 == Requirements
 
-#let mustHave = [Must Have]
-#let shouldHave = [Should Have]
-#let couldHave = [Could Have]
-#let wouldBeNiceToHave = [Would Be Nice To Have]
+#let mustHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Must Have]
+#let shouldHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Should Have]
+#let couldHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Could Have]
+#let wouldBeNiceToHave = box(fill: rgb("#cecfcf"), inset: (x: 2pt, y: 2pt), outset: (x: 2pt, y: 2pt))[Would Be Nice To Have]
+
+#let VERY_HIGH = 1
+#let HIGH = 2
+#let MEDIUM = 3
+#let LOW = 4
+#let VERY_LOW = 5
 
 #let priority(pri) = {
-	[P] + str(pri)
+	if(pri == VERY_HIGH) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt , y: 2pt), 
+				fill: rgb("#FF0407"), [P] + str(pri)) 
+		}
+	else if(pri == HIGH) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#ff390a"), [P] + str(pri)) 
+		}
+	else if(pri == MEDIUM) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#ecff09"), [P] + str(pri)) 
+		}
+	else if(pri == LOW) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#1ae53e"), [P] + str(pri)) 
+		}
+	else if(pri == VERY_LOW) { 
+			box(
+				inset: (x: 2pt, y: 2pt),
+				outset: (x: 2pt, y: 2pt), 
+				fill: rgb("#1800FF"), [P] + str(pri)) 
```

Comment:
The background and text color contrast for P5 is very low, making it hard to read.